invenio_records_rest/loaders/marshmallow.py

```diff
@@ -23,15 +23,23 @@
 from marshmallow import __version_info__ as marshmallow_version
 
 
-def _flatten_marshmallow_errors(errors):
+def _flatten_marshmallow_errors(errors, parents=[]):
```

Comment:
Well well well! I am happy that finally I have some support on fighting back mutability and go in direction of immutability :)

While I agree 100%, in this specific case I would go more for `[]` instead of None or immutable for the following reasons:
* it is much more intuitive to see a list as default when reading the signature of the method
* it is used only internally by the class `MarshmallowErrors`
* in Python, even if you use as default `None`, I can still call the method passing a var as a `list`
* the first param is already mutable, and it should not

If we really want to avoid this, I would rather declare a new inner recursive function (or not) that "hides" this extra parameter that was added, e.g.:
```
def _flatten_marshmallow_errors(errors):
    def _traverse_errors(...):
        ....
        recursive function
        ....

    res = _traverse_errors(...)
    return res
```

Reply:
In this case it's ok, since `parent` isn't actually mutated ever inside the function (e.g. no `parent.append(...)` or `parent += ...` takes place), but rather used to build a completely new list as the argument for the recursive call.

Swapping it with a tuple would e.g. completely eliminate the issue (since these wouldn't be mutable by definition)

Reply:
Yes, using a tuple makes more sense for this since we're just creating a new array of parents each recursive call anyway.